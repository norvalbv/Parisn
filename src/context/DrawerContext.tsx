import React, { createContext, ReactElement, useMemo, useState } from 'react';

export type DrawerContextProps = {
  openedId: string;
  closeDrawer: () => void;
  openDrawer: (arg: string) => void;
};

export const DrawerContext = createContext<null | DrawerContextProps>(null);

type DrawerProviderProps = {
  children: JSX.Element | JSX.Element[];
};

export const DrawerProvider = ({ children }: DrawerProviderProps): ReactElement => {
  const [openedId, setOpenedId] = useState('');

  const openDrawer = (id: string): void => {
    setOpenedId(id);
  };

  const closeDrawer = (): void => {
    setOpenedId('');
  };

  const memoisedValue = useMemo(() => {
    return {
      openedId,
      openDrawer,
      closeDrawer,
    };
  }, [openedId]);

  return <DrawerContext.Provider value={memoisedValue}>{children}</DrawerContext.Provider>;
};
