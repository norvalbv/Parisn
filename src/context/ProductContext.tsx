import React, { createContext, useState, useMemo, useEffect, ReactElement } from 'react';
import { ProductContextData, ProductInfoValues } from '../types';

type ProductContextProviderProps = {
  children?: JSX.Element;
};

const ProductContext = createContext<ProductContextData | null>(null);

export const ProductContextProvider = ({ children }: ProductContextProviderProps): ReactElement => {
  const [productInfo, setProductInfo] = useState<ProductInfoValues | null>({
    product: null,
    price: null,
    selectedSize: null,
  });

  useEffect(() => {
    const retreviedProductInfo = localStorage.getItem('savedProductInfo') || 'null';
    const parsedData = JSON.parse(retreviedProductInfo) as ProductInfoValues;
    // Checks whether all data is truthy or not.
    const truthyDataParsed = parsedData !== null && Object.values(parsedData).every((item) => item);

    if (truthyDataParsed) {
      return setProductInfo(parsedData);
    }
  }, []);

  useEffect(() => {
    localStorage.setItem('savedProductInfo', JSON.stringify(productInfo));
  }, [productInfo]);

  const memoisedValue = useMemo(
    () => ({
      productInfo,
      setProductInfo,
    }),
    [productInfo]
  );

  return <ProductContext.Provider value={memoisedValue}>{children}</ProductContext.Provider>;
};

export default ProductContext;
