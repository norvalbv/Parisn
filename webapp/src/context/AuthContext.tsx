import { CognitoUserPool } from 'amazon-cognito-identity-js';

const poolData = {
  UserPoolId: 'eu-west-2_cuz8GIlYg',
  ClientId: '3gh7ll3fdvsq6nh642g06emka6',
};

export default new CognitoUserPool(poolData);
