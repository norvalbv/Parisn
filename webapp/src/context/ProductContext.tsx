import React, { createContext, useState, useMemo, useEffect } from 'react';
import { ProductContextData, ProductInfoValues } from '../types';

type ProductContextProviderProps = {
  children?: JSX.Element;
};

const ProductContext = createContext<ProductContextData | null>(null);

export const ProductContextProvider = ({ children }: ProductContextProviderProps) => {
  const [productInfo, setProductInfo] = useState<ProductInfoValues | null>({
    product: null,
    price: null,
    selectedSize: null,
  });

  useEffect(() => {
    const retreviedProductInfo = localStorage.getItem('savedProductInfo');
    const parsedData = JSON.parse(retreviedProductInfo || 'null');
    // Checks whether all data is truthy or not.
    const truthyDataParsed = parsedData !== null && Object.values(parsedData).every((item) => item);

    if (truthyDataParsed) {
      return setProductInfo(parsedData);
    }
  }, []);

  useEffect(() => {
    localStorage.setItem('savedProductInfo', JSON.stringify(productInfo));
  }, [productInfo]);

  const memoisedValue = useMemo(
    () => ({
      productInfo,
      setProductInfo,
    }),
    [productInfo?.product]
  );

  return <ProductContext.Provider value={memoisedValue}>{children}</ProductContext.Provider>;
};

export default ProductContext;
