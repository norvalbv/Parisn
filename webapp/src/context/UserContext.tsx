import { createContext, useState, useMemo, useEffect } from 'react';
import { AuthenticationDetails, CognitoUser, CognitoUserPool } from 'amazon-cognito-identity-js';
import { UserContextInformation, UserInformation } from '../types';

const poolData = {
  UserPoolId: 'eu-west-2_cuz8GIlYg',
  ClientId: '3gh7ll3fdvsq6nh642g06emka6',
};

// const userPool = new CognitoUserPool(poolData);

type ProductContextProviderProps = {
  children?: JSX.Element;
};

const UserContext = createContext<UserContextInformation | null>(null);

export const UserInformationProvider = ({ children }: ProductContextProviderProps) => {
  const [user, setUser] = useState<UserInformation | null>({
    id: '1231287312',
    firstName: 'Benji',
    lastName: 'Norval',
    username: 'BenjiTheGreat',
    email: 'benjinorval@gmail.com',
    image:
      'https://www.google.com/url?sa=i&url=https%3A%2F%2Fwww.henley.ac.uk%2Fnews%2F2021%2Fapplied-entrepreneurship-students-seek-support-for-start-ups&psig=AOvVaw1RhLQWkTshWUg_ndVOhtN5&ust=1670189129240000&source=images&cd=vfe&ved=0CA8QjRxqFwoTCKCd3L-x3vsCFQAAAAAdAAAAABAl',
  });

  // const login = async (Username: string, Password: string) => {
  //   return await new Promise((resolve, reject) => {
  //     const user = new CognitoUser({
  //       Username,
  //       Pool: userPool,
  //     });

  //     const authDetails = new AuthenticationDetails({
  //       Username,
  //       Password,
  //     });

  //     user.authenticateUser(authDetails, {
  //       onSuccess: (data) => {
  //         console.log('on succ', data);
  //         resolve(data);
  //       },
  //       onFailure: (err) => {
  //         console.error('on fail', err);
  //         reject(err);
  //       },
  //       newPasswordRequired: (data) => {
  //         console.log('new', data);
  //         resolve(data);
  //       },
  //     });
  //   });
  // };

  useEffect(() => {
    const retreviedProductInfo = localStorage.getItem('userInformation');
    const parsedData: UserInformation = JSON.parse(retreviedProductInfo || 'null');

    setUser(parsedData ?? null);
  }, []);

  useEffect(() => {
    localStorage.setItem('userInformation', JSON.stringify(user));
  }, [user]);

  const memoisedValue = useMemo(
    () => ({
      user: {
        id: user?.id,
        firstName: user?.firstName,
        lastName: user?.lastName,
        username: user?.username,
        email: user?.email,
        image: user?.image,
      },
      setUser,
    }),
    [user]
  );

  return <UserContext.Provider value={memoisedValue}>{children}</UserContext.Provider>;
};

export default UserContext;
