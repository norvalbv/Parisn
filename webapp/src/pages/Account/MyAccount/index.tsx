import { useTransition, animated } from '@react-spring/web';
import { Dispatch, SetStateAction, useState } from 'react';
import Button from '../../../components/Button';
import { PRODUCT_1_IMAGE } from '../../../constants';
import useUser from '../../../hooks/useUser';
import MyInformation from './TabPages/MyInformation';
import MyOrders from './TabPages/MyOrders';

type MyAccountProps = {
  isOpened: { accountOpen: boolean; setAccountOpen: Dispatch<SetStateAction<boolean>> };
};

const MyAccount = ({ isOpened }: MyAccountProps) => {
  const { setUser, user } = useUser();

  const transitions = useTransition(isOpened.accountOpen, {
    from: { width: '0%', display: 'none' },
    enter: { width: '66%', display: 'block' },
    leave: { width: '0%', display: 'none' },
  });

  const [selectedTab, setSelectedTab] = useState('myinformation');

  return transitions(
    (styles, item) =>
      item && (
        <animated.div
          style={styles}
          className="w-2/3 h-screen bg-primary-dark/90 z-50 top-0 bottom-0 right-0 fixed rounded-bl-lg rounded-tl-lg"
        >
          <div className="flex items-center justify-between mb-4 p-4 border-b">
            <h2>My Account</h2>
            <Button
              text="X"
              size="xs"
              width="1.5rem"
              borderRequired="none"
              hoverColorRequired={false}
              onClick={() => isOpened.setAccountOpen(false)}
              classes="hover:rotate-90 transform-all duration-300"
            />
          </div>
          <div className="h-full grid grid-cols-3 divide-x divide-x-gray">
            <div className="p-4">
              <div className="flex gap-6 items-center border-b pb-4 mb-4">
                <img
                  src={PRODUCT_1_IMAGE}
                  alt="User Profile Picture"
                  className="rounded-full h-14 w-14 border-2"
                />
                <h3 className="text-2xl capitalize">Hello, {user.firstName}!</h3>
              </div>
              <div className="divide-y">
                <p
                  className="py-2 cursor-pointer hover:underline"
                  onClick={() => setSelectedTab('myinformation')}
                >
                  My Information
                </p>
                <p
                  className="py-2 cursor-pointer hover:underline"
                  onClick={() => setSelectedTab('myorders')}
                >
                  My Orders
                </p>
                <Button
                  text="Log Out"
                  onClick={() => {
                    setUser(null);
                    isOpened.setAccountOpen(false);
                  }}
                  navigateTo="/"
                  classes="text-left px-0 hover:underline"
                  upperCase={false}
                  width="100%"
                  borderRequired="bottom"
                  fontWeight="normal"
                  size="lg"
                  hoverColorRequired={false}
                  textOrientation="text-left"
                />
              </div>
            </div>
            <div className="p-4 w-80">
              {selectedTab === 'myinformation' ? <MyInformation /> : <MyOrders />}
            </div>
          </div>
        </animated.div>
      )
  );
};

export default MyAccount;
