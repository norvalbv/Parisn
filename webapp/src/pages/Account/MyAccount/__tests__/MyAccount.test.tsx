import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';

import MyAccount from '..';
import { UserInformationProvider } from '../../../../context/UserContext';

describe('<MyAccount />', () => {
  test('should render correctly', () => {
    const tree = renderer
      .create(
        <UserInformationProvider>
          <BrowserRouter>
            <MyAccount />
          </BrowserRouter>
        </UserInformationProvider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
