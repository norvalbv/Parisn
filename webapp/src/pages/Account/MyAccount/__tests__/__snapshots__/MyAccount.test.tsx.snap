// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MyAccount /> should render correctly 1`] = `
<button
  className="relative inline-flex items-center justify-center text-center hover:bg-buttons-hover uppercase  px-6 py-2.5 text-sm w-full my-96 mx-10 border py-4"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  role="button"
  style={
    {
      "backgroundColor": undefined,
      "color": undefined,
      "width": "25rem",
    }
  }
  type="button"
>
  <span
    className="inline-block font-semibold"
  >
    Log Out
  </span>
</button>
`;
